### PR TITLE
Add info field to query response

### LINKED ISSUE #6117: Add support to the `Info` field in the body of query API responses
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**

Starting with Prometheus 2.54.0, the response body of query APIs will have a `info` field that is used to store potential query issues, that may or may not be a false positive. To maintain Prometheus compatibility, we should support to this field in cortex as well.

## References

https://github.com/prometheus/prometheus/pull/14327
https://github.com/prometheus/prometheus/pull/14429/

### PR DESCRIPTION
**Pull request by SungJin1212**
**What this PR does**:
From Prometheus 2.54.0, an info field is added to store query issues. So, adding it to the Cortex for Prometheus compatibility is also necessary.
**Which issue(s) this PR fixes**:
Fixes #6117 

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by SungJin1212** [comment]
@rapphil @yeya24 
Could you please take a look?

File: pkg/ruler/frontend_client.go [hunk 1]
```diff
@@ -101,5 +101,9 @@ func (p *FrontendClient) InstantQuery(ctx context.Context, qs string, t time.Tim
 		level.Warn(log).Log("warnings", warning, "query", qs)
 	}
 
+	if len(infos) > 0 {
+		level.Warn(log).Log("infos", infos, "query", qs)
+	}
```
Comment (yeya24) [inline pkg/ruler/frontend_client.go:106]: I think we can remove this log as info is not warning at all. No need to log them

File: pkg/ruler/frontend_decoder.go [hunk 2]
```diff
@@ -17,45 +17,47 @@ const (
...
 
-func (j JsonDecoder) Decode(body []byte) (promql.Vector, Warning, error) {
+func (j JsonDecoder) Decode(body []byte) (promql.Vector, Warning, Info, error) {
 	var response struct {
 		Status    string          `json:"status"`
 		Data      json.RawMessage `json:"data"`
 		ErrorType string          `json:"errorType"`
 		Warnings  Warning         `json:"warnings"`
+		Infos     Info            `json:"infos"`
 		Error     string          `json:"error"`
```
Comment (yeya24) [inline pkg/ruler/frontend_decoder.go:29]: We probably missed this during the code review. Can we use https://github.com/cortexproject/cortex/blob/master/pkg/util/api/response.go#L20 struct instead?

Comment (SungJin1212) [reply pkg/ruler/frontend_decoder.go]: Let me fix it

File: pkg/ruler/frontend_decoder.go [hunk 3]
```diff
@@ -19,13 +21,8 @@ type JsonDecoder struct{}
...
 
 func (j JsonDecoder) Decode(body []byte) (promql.Vector, Warning, error) {
-	var response struct {
-		Status    string          `json:"status"`
-		Data      json.RawMessage `json:"data"`
-		ErrorType string          `json:"errorType"`
-		Warnings  Warning         `json:"warnings"`
-		Error     string          `json:"error"`
-	}
+	var response api.Response
```
Comment (yeya24) [inline pkg/ruler/frontend_decoder.go:24]: We still need to add `infos` field to `api.Response` struct

Comment (SungJin1212) [reply pkg/ruler/frontend_decoder.go:24]: I have added it.

**Review by yeya24 — APPROVED** [review]
Thanks

**Pull request merged**